zulipterminal/ui_tools/boxes.py

```diff
@@ -1118,7 +1118,19 @@ def soup2markup(
                 markup.append(("msg_code", tag_text))
             elif tag == "div" and "codehilite" in tag_classes:
                 # CODE (BLOCK?)
-                markup.append(("msg_code", tag_text))
+                pre_tag = element.contents[0]
+                code_soup = pre_tag.contents[1]
+                # NOTE: Old messages don't have the
+                # additional `code` tag.
+                if code_soup.name != "code":
+                    code_soup = pre_tag
```

Comment:
No this is only from observation

Reply:
We should definitely clarify when this occurs, or used to occur. Do you have examples of old and new messages exhibiting this? Given we can look at the source of them, is this literally for messages before some date, or is it a different style of markdown?

Reply:
Examples:
* https://chat.zulip.org/#narrow/stream/7-test-here/topic/code/near/130791
* https://chat.zulip.org/#narrow/stream/7-test-here/topic/html.20elements.20in.20code.20blocks/near/730016
Possibly before April 2019 atleast
Example of same message before and after:
```html
<pre><span></span>&amp;lt;span&amp;gt;test&amp;lt;/span&amp;gt;
</pre>

<pre><span></span><code>&amp;lt;span&amp;gt;test&amp;lt;/span&amp;gt;
</code></pre>
```

Reply:
It would be good to get a point of reference for this - maybe a czo discussion exists? The server changelog or commits may include this, but it may be hard to find, so start a new topic on eg #backend if you can't find a reference.